frontend/public/components/storage/shared.ts

```diff
@@ -58,8 +123,64 @@ export const dropdownUnits = {
   Gi: 'GiB',
   Ti: 'TiB',
 };
-export const getAccessModeForProvisioner = (provisioner: string) => {
-  return provisioner && isCephProvisioner(provisioner)
-    ? ['ReadWriteOnce', 'ReadWriteMany']
-    : ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'];
+
+const getAllowedCombiationsForProvisioner = (provisioner: string): AllowedCombinations => {
+  const combinationsForProvisioner: AllowedCombinations =
+    provisionerAccessModeMapping[provisioner] || [];
+  return combinationsForProvisioner.length && !combinationsForProvisioner.includes(['All'])
+    ? combinationsForProvisioner
+    : allowedCombinations.filter((combination) => !combination.includes('All'));
+};
+
+const filterAllowedCombiations = (
+  combinationsForProvisioner: AllowedCombinations,
+  mode: string,
+  value: string,
+): AllowedCombinations => {
+  return value
+    ? mode === 'volumeMode'
+      ? combinationsForProvisioner.filter((combination) => combination[1] === value)
+      : combinationsForProvisioner.filter((combination) => combination[0] === value)
```

Comment:
Avoid nested ternaries as they are hard to understand.

I'm a little unclear on the logic here. I think this is confusing because we're trying to combine too many things in a single nested array instead of having separate arrays.